core/src/main/java/com/netflix/conductor/core/execution/tasks/DoWhile.java

```diff
@@ -0,0 +1,168 @@
+		List<WorkflowTask> loopOver = task.getWorkflowTask().getLoopOver();
+
+		for(WorkflowTask workflowTask : loopOver){
+			String refName = getTaskRefName(workflowTask, task);
+			Task loopOverTask = workflow.getTaskByRefName(refName);
+			if(loopOverTask == null){
+				//Task is not even scheduled yet
+				allDone = false;
+				break;
+			}
+			Status taskStatus = loopOverTask.getStatus();
+			hasFailures = !taskStatus.isSuccessful();
+			if(hasFailures){
+				failureReason.append(loopOverTask.getReasonForIncompletion()).append(" ");
+			}
+			task.getOutputData().put(workflowTask.getTaskReferenceName(), loopOverTask.getOutputData());
+			allDone = taskStatus.isTerminal();
+			if(!allDone || hasFailures){
+				break;
+			}
+		}
+		if (hasFailures) {
+			return markLoopTaskFailed(task, failureReason.toString());
+		} else if (!allDone) {
+			return false;
+		}
+		boolean shouldContinue = getEvaluatedCondition(task, workflow);
+		logger.debug("taskid {} condition evaluated to {}",task.getTaskId(), shouldContinue);
+		if (shouldContinue){
+			return scheduleLoopTasks(task, workflow, provider);
+		} else {
+			return markLoopTaskSuccess(task);
+		}
+	}
+
+	private String getTaskRefName(WorkflowTask workflowTask, Task task) {
+		String refName = workflowTask.getTaskReferenceName();
+		if (task.getIteration()>0) {
+			refName +=  "_" + task.getIteration();
+		}
+		return refName;
+	}
+
+	Map<String, Object> getConditionInput(Task task, Workflow workflow) {
+		Map<String, Object> map = new HashMap<>();
+		List<WorkflowTask> loopOverTask = task.getWorkflowTask().getLoopOver();
+		for (WorkflowTask workflowTask : loopOverTask) {
+			Task temp = workflow.getTaskByRefName(getTaskRefName(workflowTask, task));
+			map.put(workflowTask.getTaskReferenceName(), temp.getOutputData());
+			logger.debug("Taskname {} output {}", temp.getReferenceTaskName(), temp.getOutputData());
+		}
+		return map;
+	}
+
+	boolean scheduleLoopTasks(Task task, Workflow workflow, WorkflowExecutor provider) {
+		logger.debug("Scheduling loop tasks for taskid {} as condition {} evaluated to true",
+				task.getTaskId(), task.getWorkflowTask().getLoopCondition());
+		List<WorkflowTask> loopOver = task.getWorkflowTask().getLoopOver();
+		List<Task> taskToBeScheduled = new ArrayList<>();
+		int iteration = task.getIteration() + 1;
+		for(WorkflowTask loopOverRef : loopOver){
+			String refName = getTaskRefName(loopOverRef, task);
+			Task existingTask = workflow.getTaskByRefName(refName);
+			existingTask.setRetried(true);
+			Task newTask = provider.taskToBeRescheduled(existingTask);
+			newTask.setSeq(existingTask.getSeq());
+			newTask.setReferenceTaskName(loopOverRef.getTaskReferenceName() + "_" + iteration);
+			newTask.setRetryCount(existingTask.getRetryCount());
+			newTask.setScheduledTime(System.currentTimeMillis());
+			taskToBeScheduled.add(newTask);
+		}
+		task.setIteration(iteration);
+		provider.scheduleTask(workflow, taskToBeScheduled);
```

Comment:
I have to make scheduleTask method public so that task will be scheduled on immediate basis here. I feel this violation of separation of concern. Task execute method should not schedule another task.

So I tried different approach to come up with loop. I changed https://github.com/manan164/conductor/blob/do-while-conductor/core/src/main/java/com/netflix/conductor/core/execution/mapper/DoWhileTaskMapper.java#L48 method and evaluate condition. If that condition is true then schedule the task again and if that false do not schedule. Decision task does that.

Can you people tell what is the good way of doing this. @naveenchlsn @mudit3774 @kishorebanala @apanicker-nflx @shivambansal2222

Reply:
IMO, do not use the `READY_FOR_RERUN` status. We would like to deprecate/remove this status in an upcoming version. 
Are you storing the history of past executions of the loop task?

Reply:
Hi @apanicker-nflx Thanks for the reply. I have changed my approach. Will add in this pr after sufficient dev testing. Now I am deleting execution history of each task after terminal execution and store the output of each iteration in do while task as iteration as key. The approach mentioned in this PR will not work for loopOver task like fork join and decision because I was changing task reference name. I have to maintain original taskRefName in order to get task mapper work properly in second iteration. If storing history is concern then I think while storing I can append iteration.